app/javascript/dashboard/components/ui/TimeAgo.vue

```diff
@@ -28,51 +26,45 @@ export default {
   },
   data() {
     return {
-      timeAgo: '',
+      timeAgo: this.dynamicTime(this.timestamp),
       timer: null,
     };
   },
-  computed: {
-    refreshTime() {
-      const timeDiff = differenceInMilliseconds(
-        new Date(),
-        new Date(this.timestamp * 1000)
-      );
-      if (timeDiff > DAY_IN_MILLI_SECONDS) {
-        return DAY_IN_MILLI_SECONDS;
-      }
-      if (timeDiff > HOUR_IN_MILLI_SECONDS) {
-        return HOUR_IN_MILLI_SECONDS;
-      }
-      if (timeDiff > MINUTE_IN_MILLI_SECONDS) {
-        return MINUTE_IN_MILLI_SECONDS;
-      }
-      return ZERO;
+  watch: {
+    timestamp() {
+      this.timeAgo = this.dynamicTime(this.timestamp);
     },
   },
   mounted() {
-    this.timeAgo = this.dynamicTime(this.timestamp);
     if (this.isAutoRefreshEnabled) {
       this.createTimer();
     }
   },
   beforeDestroy() {
-    this.clearTimer();
+    clearInterval(this.timer);
```

Comment:
@KubesDavid Can you please rewrite this with `setTimeout` instead of `setInterval`? We had some performance issues previously with missing `clearInterval` calls. It is quite easy to forget and can miss this in the review as well. We would try to use only setTimeout and probably add a linter for the same.